shared/macros-bash.jinja

```diff
@@ -7,3 +56,95 @@
 {{% macro die(message, rc=1, action="exit") -%}}
 { printf '%s\n' '{{{ message }}}' >&2; {{{ action }}} {{{ rc }}}; }
 {{%- endmacro %}}
+
+{{#
+# Add an entry to a text configuration file
+# path: path of the configuration file
+# parameter: the paramenter to be set in the configuration file
+# value: the value of the parameter to be set in the configuration file
+# create: whether create the file specified by path if the file does not exits
+# insert_after: inserts the entry right after first line that matches regular expression specified by this argument, set to EOF to insert at the end of the file
+# insert_before: inserts the entry right before first line that matches regular expression specified by this argument, set to BOF to insert at the beginning of the file
+# insensitive: ignore case
+# separator: separates parameter from the value (literal)
+# separator_regex: regular expression that describes the separator and surrounding whitespace
+# prefix_regex: regular expression describing allowed leading characters at each line
+#}}
+{{%- macro set_config_file(path, parameter, value, create, insert_after, insert_before, insensitive, separator, separator_regex, prefix_regex) -%}}
+    {{%- set line_regex = prefix_regex+parameter+separator_regex -%}}
+    {{%- set new_line = parameter+separator+value -%}}
+    {{{ lineinfile(path, line_regex, "", false, false, "", "", insensitive) }}}
+    {{{ lineinfile(path, "", new_line, create, true, insert_after, insert_before, insensitive) }}}
+{{%- endmacro -%}}
+
+{{%- macro lineinfile(path, regex, line, create=false, present=true, insert_after="", insert_before="", insensitive=true) -%}}
+    {{%- if not present -%}}
+        if [ -e "{{{ path }}}" ] ; then
```

Comment:
The previous `-%}}` eat the leading whitespace from `... if`, but there is nothing eaten from the corresponding `... fi`.